lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3069,6 +3069,17 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
       ->castTo<AnyFunctionType>();
   bool isMethod = original->hasImplicitSelfDecl();
 
+  // Check if there are any inout parameters.
+  for (auto &param : originalFnTy->getParams()) {
```

Comment:
Since you need to check for with-respect-to-ness, this step should be done after checked parameter indices are formed later in this function.

Reply:
I moved the check inside `checkDifferentiationParameters`. Will that do it?